src/lsstseries/analysis/structure_function/iqr/calculator.py

```diff
@@ -0,0 +1,123 @@
+from typing import List
+
+import numpy as np
+from scipy.stats import binned_statistic
+
+from lsstseries.analysis.structure_function.base_argument_container import StructureFunctionArgumentContainer
+from lsstseries.analysis.structure_function.base_calculator import StructureFunctionCalculator
+
+# For details see Kozlowski 16 Equation 10: https://arxiv.org/abs/1604.05858
+CONVERSION_TO_SIGMA = 0.741
+
+
+class IqrStructureFunctionCalculator(StructureFunctionCalculator):
+    """This class implements the structure function calculation described in
+    Equation 10 of Kozlowski 16: https://arxiv.org/abs/1604.05858
+
+    SF_obs(deltaT) = 0.741 * IQR
+
+    Where `IQR` is the interquartile range between 25% and 75% of the sorted
+    (y(t) - y(t+delta_t)) distribution.
+    """
+
+    def __init__(
+        self,
+        time: List[List[float]],
+        flux: List[List[float]],
+        err: List[List[float]],
+        argument_container: StructureFunctionArgumentContainer,
+    ):
+        super().__init__(time, flux, err, argument_container)
+
+    def calculate(self):
```

Comment:
In a follow up PR, I would like to abstract the majority of the work here in `calculate` into the parent class. It seems like each calculation method is going to require very similar data manipulation at this step, and I believe that the majority of the code in lines 33-57 could be one parent class method, and then lines 60-73 could be another parent class method that would accept an input argument that defined what statistic to use in `binned_statistic`.

Reply:
Created an issue for this here: https://github.com/lincc-frameworks/lsstseries/issues/103